### PR TITLE
Add Information About Building a Snapshot Version

### LINKED ISSUE #11027: Add Information to CONTRIBUTING.md about JHipster Snapshot Versions
##### **Overview of the issue**

It's my understanding that if we are using a snapshot dependency version we have to build the [jhipster/jhipster](https://github.com/jhipster/jhipster) project so that the generator correctly points to the snapshot. I suggest adding this information to the CONTRIBUTING.md especially for new users who are setting up their development environment for the first time. 

##### **Motivation for or Use Case**

Save development time and ease of quickly setting up the dev environment for new users. 

##### **Reproduce the error**

Change the [`jhipster-dependencies.version` line in `pom.xml.ejs`](https://github.com/jhipster/generator-jhipster/blob/257a56d1e787cadb8443a6b9d5f2bd19ba7952b5/generators/server/templates/pom.xml.ejs#L70) to a snapshot version (if it's not already on a snapshot version), remove the contents of the .m2 directory and try to generate and build a project using the generator. 

##### **Related issues**

https://github.com/jhipster/generator-jhipster/issues/9571

##### **Suggest a Fix**

Add a note to the CONTRIBUTING.md file about cloning the jhipster/jhipster project and building it in case of snapshot release. 

-   [x] Checking this box is mandatory (this is just to show you read everything)

### PR DESCRIPTION
**Pull request by SudharakaP**
Fixes #11027

-   Please make sure the below checklist is followed for Pull Requests.

-   [x] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [x] Tests are added where necessary
-   [x] Documentation is added/updated where necessary
-   [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -255,6 +255,13 @@ jhipster --with-entities
 
 You should see your changes reflected in the generated project.
 
+Note: The generated project might not build properly in case the generator is using a
+snapshot version of [jhipster/jhipster](https://github.com/jhipster/jhipster). This issue is mentioned in; https://github.com/jhipster/generator-jhipster/issues/9571. In
+this case clone the jhipster/jhipster project and build it using: 
+```
+./mvnw clean install -Dgpg.skip=true`
```
Comment (pascalgrimaud) [inline CONTRIBUTING.md]: can you remove the last char `

Comment (SudharakaP) [reply CONTRIBUTING.md]: @pascalgrimaud : Good catch. Done :smile:

**Comment by SudharakaP** [comment]
@kaidohallik : Thanks. Good point; I've added it :+1:

**Comment by SudharakaP** [comment]
@kaidohallik : You are welcome :smile:

**Pull request merged**